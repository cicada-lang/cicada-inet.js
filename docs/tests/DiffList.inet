defnode sole -- value! end

defnode null -- value! end

defnode cons
  tail
  head
  -------
  value!
end

defnode diff
  left
  right
  ------
  value!
end

defnode diff_append
  rest
  target!
  ------
  return
end

defrule diff diff_append
  (diff)-left (diff_append)-rest diff_open
  (diff)-right diff
  return-(diff_append)
end

defnode diff_open
  list
  target!
  ------
  return
end

defrule diff diff_open
  (diff_open)-list (diff)-left connect
  (diff)-right
  return-(diff_open)
end

defnet one_two_soles
  wire sole cons diff
  wire sole cons sole cons diff
  diff_append
end

defnet two_two_soles
  wire sole cons sole cons diff
  wire sole cons sole cons diff
  diff_append
end

show one_two_soles end
run one_two_soles end

show two_two_soles end
run two_two_soles end
