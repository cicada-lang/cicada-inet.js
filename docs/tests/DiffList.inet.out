(diff₉)-value!target-(diff_append₁₀)
(diff₉)-left tail-(cons₆)
(diff₃)-left tail-(cons₂)
(diff₃)-value rest-(diff_append₁₀)
(cons₈)-value right-(diff₉)
(cons₆)-value tail-(cons₈)
(sole₇)-value head-(cons₈)
(sole₅)-value head-(cons₆)
(cons₂)-value right-(diff₃)
(sole₁)-value head-(cons₂)
(diff_append₁₀)-return

(cons₁₃)-value left-(diff₂₄)
(cons₁₃)-tail tail-(cons₁₇)
(cons₁₉)-value right-(diff₂₄)
(cons₁₇)-value tail-(cons₁₉)
(sole₁₈)-value head-(cons₁₉)
(sole₁₆)-value head-(cons₁₇)
(sole₁₂)-value head-(cons₁₃)
(diff₂₄)-value

(diff₃₆)-value!target-(diff_append₃₇)
(diff₃₆)-left tail-(cons₃₃)
(diff₃₀)-left tail-(cons₂₇)
(diff₃₀)-value rest-(diff_append₃₇)
(cons₃₅)-value right-(diff₃₆)
(cons₃₃)-value tail-(cons₃₅)
(sole₃₄)-value head-(cons₃₅)
(sole₃₂)-value head-(cons₃₃)
(cons₂₉)-value right-(diff₃₀)
(cons₂₇)-value tail-(cons₂₉)
(sole₂₈)-value head-(cons₂₉)
(sole₂₆)-value head-(cons₂₇)
(diff_append₃₇)-return

(cons₄₂)-value left-(diff₅₃)
(cons₄₀)-tail tail-(cons₄₆)
(cons₄₈)-value right-(diff₅₃)
(cons₄₆)-value tail-(cons₄₈)
(sole₄₇)-value head-(cons₄₈)
(sole₄₅)-value head-(cons₄₆)
(cons₄₀)-value tail-(cons₄₂)
(sole₄₁)-value head-(cons₄₂)
(sole₃₉)-value head-(cons₄₀)
(diff₅₃)-value

